app/controllers/notifications_controller.rb

```diff
@@ -19,12 +19,15 @@ def index
       num = @initial_page_size
     end
 
+    logger.debug "notification setting: #{@user.notification_setting.inspect}"
     @notifications = if (params[:org_id].present? || params[:filter] == "org") && allowed_user?
                        organization_notifications
                      elsif params[:org_id].blank? && params[:filter].present?
                        filtered_notifications
-                     else
+                     elsif @user.notification_setting.new_post_notifications?
```

Comment:
Instead of filtering notifications before displaying them, we should avoid creating notifications in the first place.
To do this we need to adjust finding article followers in `Notifications::NotifiableAction::Send` : exclude users with the corresponding notification setting set to `false`.